docroot/modules/custom/foia_file/drush/foia_file.drush.inc

```diff
@@ -0,0 +1,174 @@
+<?php
+
+/**
+ * @file
+ * FOIA Api module drush command.
+ */
+
+use Drupal\file_entity\Entity\FileEntity;
+
+/**
+ * Implements hook_drush_command().
+ */
+function foia_file_drush_command() {
+  $items = [];
+
+  $items['file-entity-update'] = [
+    'description' => 'Updates file entities virus scan statuses.',
+    'core' => ['8+'],
+    'arguments' => [
+      'scan_result' => 'The result of the file scan.',
+    ],
+    'aliases' => ['feu'],
+  ];
+
+  return $items;
+}
+
+/**
+ * Implements hook_drush_help().
+ */
+function foia_file_drush_help($command) {
+  switch ($command) {
+    case 'drush:file-entity-update':
+      return dt('Update virus scan statuses for file entites based upon output from virus scanner.');
+  }
+}
+
+/**
+ * Implements drush_hook_COMMAND().
+ */
+function drush_foia_file_file_entity_update($scanOutput) {
+  $startTime = microtime(TRUE);
+
+  // Get webform directory on the server.
+  $fileDir = explode('/webform/', $scanOutput, 2);
+  $webformDir = "{$fileDir[0]}/webform/";
+
+  // Temporarily delete similar paths of the webform dir during processing.
+  $trimmedFiles = str_replace($webformDir, '', $scanOutput);
+
+  // Split string into lines. @todo change to preg split \R?
+  $scans = explode("\n", $trimmedFiles);
+
+  foreach ($scans as $scan) {
+
+    // Split the scan line into filename and status.
+    preg_match('/[^ ]*$/', $scan, $wordsArray);
+    // Set status to the last word of the line.
+    $status = $wordsArray[0];
+
+    $expectedStatuses = ['OK', 'Removed.', 'Empty file'];
+    if (!in_array($status, $expectedStatuses)) {
+      \Drupal::logger('foia_file')->warning("An unexpected status was detected in the virus scan output.");
+    }
+
+    // Get relative file path.
+    $scanLength = strlen($scan);
+    $statusLength = strlen($status);
+    // Length of the file name is the remainder of the scan minus the colon and
+    // space characters.
+    $fileNameLength = $scanLength - $statusLength - 2;
+    $fileName = substr($scan, 0, $fileNameLength);
+
+    // Validate path, filename, and extension.
+    $pathParts = pathinfo($fileName);
+    foreach ($pathParts as $pathPart) {
+      if (empty($pathPart)) {
+        $status = 'invalid';
+      }
+    }
```

Comment:
I'm not sure this check is necessary, since we don't really act on `invalid` statuses. We're probably also not 100% familiar with how this function behaves, so I'm hesitant to add in a check using it unless we know exactly what we're looking for. A file might not have an extension, for example, so that path part might be empty. I'm not sure we'd want to consider that invalid. We could just try to look up the corresponding file entity regardless.